splinterd/src/config/mod.rs

```diff
@@ -425,6 +425,24 @@ impl Config {
             );
         }
     }
+
+    fn log_config_directories(&self) {
+        debug!(
+            "Config: config_dir: {:?} (source: {:?})",
+            self.config_dir(),
+            self.config_dir_source()
+        );
+        debug!(
+            "Config: tls_cert_dir: {} (source: {:?})",
+            self.tls_cert_dir(),
+            self.tls_cert_dir_source()
+        );
+        debug!(
+            "Config: state_dir: {} (source: {:?})",
+            self.state_dir(),
+            self.state_dir_source()
+        );
+    }
 }
```

Comment:
Seems unnecessary to split this out from the main function

Reply:
There was a clippy error when I added the whitelist debug statement about the complexity of the function being too high. This was an attempt to not run into that clippy error again.

Reply:
Hmm, this might actually be a case where ignoring the lint is the right thing to do. The cognitive complexity of the `log_as_debug` function is actually very low since it only does one thing; it just does it many times. IMO splitting parts of it out just makes it more complex, because you have to look at the various sub-functions to get the full picture.

Reply:
Mmkay, ya I would agree. I'll fix that